### DISCUSSION TITLE
(Paid) Looking for an Expert on Webkit debugging / Model & Texture Quality

### DISCUSSION THREAD

**Discussion by phreak** [google/model-viewer#4850] [General]
Hi all,

We encountered a peculiar issue causing Webkit browsers to crash all 1-5 times when loading a Model Viewer (MV) model (using compressed GLBs) from buttons inside the MV (configurator-style setup). The first model appears fine, but switching colours or to another model crashes the browser.

Initially, the browser attempts to reload the page. Eventually, it goes blank with the message: "A problem repeatedly occurred on 'https://...'."

Simultaneously, Quick Look AR crashes or loads endlessly every time we open AR mode on iPhones, where we use dedicated USDZs (no auto-convert).

Safari's Technology Preview is also able to help us partly but not really by identifying the bug. Sometimes a worker starts (WebP decompression?), sometimes not, right before the browser crashes.

We're well-versed in Blender and Model Viewer, having set up over 100 (working) viewers in the past, but this issue has us perplexed. Neither our good knowledge of modelling/textures/web development or integrating KTX2 nor MeshOptim compression resolves the problem.

I cannot share the link here publicly. If you're an expert, particularly someone who has contributed to the Model Viewer project before, we'd appreciate your help as a paid consultant. Please contact me at traik@polarisedkids.com.

Kindly,
phreak

**Comment by phreak** [comment]
After another day of testing, we are pretty sure that a newer iOS version introduced another memory limit or some sort of bug about decompressing textures or meshes. We experience major issues among many projects of different natures, which lead us to reduce texture quality/dimensions/alpha's, to close to make things work.

If you experience similar let us know. 

ARkit and Quick Look are not what they are supposed to be, a productive and professional environment for XR. From documentation, to release notes of their tools, a broken Safari Technology Preview, and no proper error handling in AR Quick Look leaving developers and end users confused, the latest AR Quick Look also often loses focus and shows blurred models... it's a pity.

**Reply by elalish** [reply]
@AdaRoseCannon to ensure Apple gets this feedback.

**Reply by AdaRoseCannon** [reply]
Please use Feedback Assistant to report the bug and include URLs to the generated USD files which present the issue.

**Reply by phreak** [reply]
Sure, we posted feedback there for various cases. Now since working with ARkit for many years, I have the feeling that Apple's AR/VR/XR unit barely exists OR the directive is still "They should make an app!" 😓 Sorry, when there is nothing left to do, frustration comes in handy when silver fishes explain how great Apple is in their Apple Events, while whole tech stacks collapse below.
